### PR TITLE
Script to compress bit-shaved netcdf output 

### PR DESCRIPTION
**Pull request by gmao-rreichle**
This PR adds a utility to compress bit-shaved netcdf output. 

Original version provided by @gmao-qliu, lightly edited by @gmao-rreichle.

@weiyuan-jiang: Please review and add multi-threading capability.

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
@gmao-qliu , I thought we can use the same name as input and output. Can you test my changes?

**Comment by gmao-qliu** [comment]
> 

it's unsafe to use the same name for input and output according to the nco user guide.

**Comment by weiyuan-jiang** [comment]
> it's unsafe to use the same name for input and output according to the nco user guide.

I think nco internally did the same thing as you did.

File: src/Applications/LDAS_App/util/compress_nc.sh
Comment (gmao-rreichle) [inline src/Applications/LDAS_App/util/compress_nc.sh:22]: @weiyuan-jiang:  I'm concerned the ampersand at the end of the `ncks` command will create problems if there are many files in the directory.  If there are, say, 100 nc4 files, then 100 ncks commands would be kicked off nearly instantaneously, which may bog down the login node.  For multi-threading, I think we would need a more sophisticated approach that limits the number of simultaneous ncks jobs to, say, 50% of the number of available processors.  I imagine this isn't trivial, and it may be simpler just to run the ncks jobs sequentially.  Thoughts?

**Comment by gmao-rreichle** [comment]
> > it's unsafe to use the same name for input and output according to the nco user guide.
> 
> I think nco internally did the same thing as you did.

@gmao-qliu: Is it ok with you to use the same file name going into and out of ncks?

**Comment by gmao-qliu** [comment]
> 

It's fine with me. Seems it's only unsafe with the --no_tmp_fi flag. @gmao-rreichle @weiyuan-jiang

File: src/Applications/LDAS_App/util/compress_nc.sh
Comment (weiyuan-jiang) [reply src/Applications/LDAS_App/util/compress_nc.sh]: usually if a users has many jobs, he should not do it on login nodes.  If time is not an issue, sequential jobs is totally fine.

**Comment by gmao-rreichle** [comment]
@weiyuan-jiang, @biljanaorescanin:  I've finished my modifications of this PR but I can't approve it because I initiated it.  When you get a chance, please approve and merge, or let me know if you have any concerns.

**Review by biljanaorescanin — APPROVED** [review]

**Pull request merged**
